controllers/mattermost/mattermost/mattermost.go

```diff
@@ -334,12 +350,13 @@ func (r *MattermostReconciler) updateMattermostDeployment(
 		defer r.cleanupUpdateJob(job, reqLogger)
 	}
 	if err != nil {
-		return err
+		recStatus.ResourcesReady = false
+		return recStatus, err
```

Comment:
I believe we have an issue here. Although we are setting `ResourcesReady` to false and passing it back up with `recStatus`, we are also still passing up the error value. If I am not mistaken that error will make it all the way back to the `Reconcile` method where it will always be checked first meaning `recStatus` goes unused.

Reply:
I agree with Gabe, we need to adjust the logic here. It looks like `checkUpdateJob` will only return non `nil` job if it is completed, this should give us information on whether we set the `ResourcesReady` to true or false.

We should probably add some test to verify it.

Reply:
Yes I think that would make sense, however we would need to set it to `true` not `false` in that case, as `if job != nil` it means it completed successfully 🙂

Reply:
cool thanks

Reply:
Sorry, it looks like I can be a little more clear on what I was originally trying to point out. So, if I am following this correctly then `checkUpdateJob` will return an error unless the whole thing completes successfully. Now, the code I linked here returns both the `checkUpdateJob` err value as well as recStatus with `ResourcesReady=false`.

The issue I see here is that the `Reconcile` method above has this:

```
	recStatus, err := r.checkMattermost(mattermost, dbConfig, fileStoreConfig, &status, reqLogger)
	if err != nil {
		r.updateStatusReconcilingAndLogError(mattermost, status, reqLogger, err)
		return reconcile.Result{}, err
	}

	if !recStatus.ResourcesReady {
		reqLogger.Info("Mattermost resources not ready, delaying for 10 seconds!")
		return ctrl.Result{RequeueAfter: resourcesReadyDelay}, nil
	}
```
	
Because an error is always returned it will hit that logic fist and never get to the `ResourcesReady` check. This means we will still run into the issue noted in the linked issue.

> From reconciler perspective, when the job is still running the error is returned to requeue reconciliation request. If error occur enough number of times, in rare cases this may significantly delay rolling out the deployment due to error reconciliation back off.

Szymon, let me know if I am missing something here or looking at this the wrong way.

Reply:
I see what you are saying, yes I think you are right, we would need to adjust this part https://github.com/mattermost/mattermost-operator/blob/master/controllers/mattermost/mattermost/mattermost.go#L378-L396 and probably here https://github.com/mattermost/mattermost-operator/blob/master/controllers/mattermost/mattermost/mattermost.go#L362 to not return errors when job was created but not finished but instead set `ResourceReady` to `false`.

That is way I think we need some tests for that, we could add it here https://github.com/mattermost/mattermost-operator/blob/master/controllers/mattermost/mattermost/mattermost_test.go#L283 with something simple such as:
```go
// Check deployment - update job is not completed, therefore resource is not ready
recResult, err = reconciler.checkMattermostDeployment(mm, dbInfo, fileStoreInfo, currentMMStatus, logger)
require.NoError(t, err)
assert.False(t, recResult.ResourceReady)
```

Reply:
Yeah understood we certainly do not have to return error as mentioned by @Szymongib. Thanks for reviewing this @gabrieljackson @Szymongib :)